lib/Backends.hs

```diff
@@ -28,13 +32,30 @@ data SomeBackend where
 instance Show SomeBackend where
   show (SomeBackend a) = show a
 
+data SupportedBackend = SupportedBackend
+  { bType :: String
+  , bFromJSON :: A.Value -> Parser SomeBackend
+  , bToml :: Toml.TomlEnv SomeBackend
+  , bPathHelpMsg :: Pretty.Doc
+  }
+
+supportedBackendsMap :: HS.HashMap String SupportedBackend
+supportedBackendsMap = HS.fromList [("vault-kv", vaultKVSup)]
+  where
+    vaultKVSup = SupportedBackend
+      "vault-kv"
+      ((fmap SomeBackend) . (A.parseJSON @VaultKvBackend))
+      (fmap SomeBackend . Toml.codecRead (_codec @VaultKvBackend))
+      "Vault Kv paths can contain only the following characters: [a-zA-Z0-9] and symbols '-', '_'."
```

Comment:
Here as well: we should use `kvPathSegmentAllowedCharacters` to generate this message

Reply:
done